### PR TITLE
feat: add `color` prop to `Alert`, `Badge`, `Button`, `Chip`, `ChipGroup`, `Counter`, `Link`, & `Spinner`

### PR DESCRIPTION
**Pull request by snitin315**
### [Component Updates Doc](https://docs.google.com/document/d/1jp-3pDV62qdFvYsco8YqcUluT3kXIUIEmXBZ11ta7UA/edit?usp=sharing)

- Added new white tokens.
- Added `color` prop to components and deprecated `intent`/`variant` prop wherever applicable.

Screenshot 2023-09-25 at 9 35 03 AM

### CODE REVIEW

File: packages/blade/src/components/Button/BaseButton/BaseButton.tsx [hunk 1]
```diff
@@ -188,29 +190,33 @@ const getProps = ({
...
   size: NonNullable<BaseButtonProps['size']>;
   variant: NonNullable<BaseButtonProps['variant']>;
-  intent: BaseButtonProps['intent'];
-  contrast: NonNullable<BaseButtonProps['contrast']>;
+  color: BaseButtonProps['color'];
 }): BaseButtonStyleProps => {
+  if (variant === 'tertiary' && color !== 'default' && color !== 'white') {
+    throwBladeError({
+      moduleName: 'BaseButton',
+      message: `Tertiary buttons can only be used with default or white color. You passed ${color}`,
```
Comment (kamleshchandnani) [inline packages/blade/src/components/Button/BaseButton/BaseButton.tsx:198]:
```suggestion
      message: `Tertiary variant can only be used with color: default or white but received ${color}`,
```

File: packages/blade/src/components/Button/BaseButton/StyledBaseButton.web.tsx [hunk 2]
```diff
@@ -27,17 +27,23 @@ const StyledBaseButton = styled.button
...
     },
     '&:active': {
       backgroundColor: props.activeBackgroundColor,
-      borderColor: props.activeBorderColor,
+      ...(props.variant !== 'tertiary' && {
+        borderColor: props.activeBorderColor,
+      }),
     },
-    '&:focus': {
+    '&:focus-visible': {
```
Comment (kamleshchandnani) [inline packages/blade/src/components/Button/BaseButton/StyledBaseButton.web.tsx:40]: why did we add focus visible?

File: packages/blade/src/components/Button/BaseButton/__tests__/BaseButton.web.test.tsx [hunk 3]
```diff
@@ -152,228 +164,31 @@ describe('<BaseButton />', () => {
...
-      <BaseButton intent="notice" contrast="high" isDisabled={true}>
-        {buttonText}
-      </BaseButton>,
-    );
-    expect(container).toMatchSnapshot();
-  });
+  colors.forEach((color) => {
+    variants.forEach((variant) => {
+      // We support only white and default color for tertiary variant
+      if (variant === 'tertiary' && color !== 'white' && color !== 'default') return;
```
Comment (kamleshchandnani) [inline packages/blade/src/components/Button/BaseButton/__tests__/BaseButton.web.test.tsx:170]: can we add an assertion which throws error when someone passes the tertiary for feedback colors?

File: packages/blade/src/components/Alert/Alert.tsx [hunk 4]
```diff
@@ -212,8 +222,8 @@ const Alert = ({
       <BaseButton
         size={textSize}
         onClick={actions.primary.onClick}
-        intent={intent}
-        contrast={contrast}
+        color={color || intent}
+        variant={contrast === 'high' ? 'primary' : 'secondary'}
```
[the remark this answers is not in the document]
Comment (snitin315) [reply packages/blade/src/components/Alert/Alert.tsx:226]: It's not on the BaseButton, we have it on `Alert`.

File: packages/blade/src/components/Alert/Alert.tsx [hunk 5]
```diff
@@ -242,7 +252,11 @@ const Alert = ({
       marginRight="spacing.4"
       display={isReactNative ? castNativeType('flex') : castWebType('inline-flex')}
     >
-      <BaseLink size={textSize} contrast={contrast} intent={intent} {...secondaryActionParams}>
+      <BaseLink
+        size={textSize}
+        color={contrast === 'high' ? 'white' : intent}
```
[the remark this answers is not in the document]
Comment (snitin315) [reply packages/blade/src/components/Alert/Alert.tsx:257]: It's not on `BaseLink`, it's on `Alert`

File: packages/blade/src/components/Badge/__tests__/Badge.native.test.tsx [hunk 6]
```diff
@@ -173,6 +183,28 @@ describe('<Badge />', () => {
     expect(toJSON()).toMatchSnapshot();
   });
 
+  colors.forEach((color) => {
+    it(`should render low contrast ${color} color Badge`, () => {
```
[the remark this answers is not in the document]
Comment (snitin315) [reply packages/blade/src/components/Badge/__tests__/Badge.native.test.tsx:187]: Yes, we need it.

File: packages/blade/src/components/Badge/Badge.tsx [hunk 7]
```diff
@@ -23,11 +23,18 @@ type BadgeProps = {
...
+   * @deprecated Use the `color` prop instead
    * @default 'neutral'
    */
   variant?: Feedback | 'blue';
+  /**
+   * Sets the color of the badge.
+   *
+   * @default 'neutral'
+   */
+  color?: Feedback | 'blue';
```
[the remark this answers is not in the document]
Comment (snitin315) [reply packages/blade/src/components/Badge/Badge.tsx]: Changed to `default`.

File: .changeset/gold-toes-return.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+- **Alert** - Deprecated `intent` prop.
+- **Badge** - Deprecated `variant` prop.
+- **Chip & ChipGroup** - Deprecated `intent` prop.
+- **Counter** - Deprecated `variant` and `intent` prop.
+- **Spinner** - Deprecated the `contrast` prop.
+
+#### Jest Snapshots
+
+Your existing component snapshots may update, there is no change in the functionality of components. Feel free to commit the updated the snapshots.
```
Comment (snitin315) [inline .changeset/gold-toes-return.md:21]:
```suggestion
Your existing component snapshots may update, there is no change in the visual appearance of components. Feel free to commit the updated snapshots.
```

**Pull request merged**